pkg/controller/core/workload_controller.go

```diff
@@ -204,6 +204,7 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			wl.Status.RequeueState = nil
 			updated = true
 		}
+		updated = updated || workload.ResetChecksOnDeactivation(&wl)
```

Comment:
swap the order: ` workload.ResetChecksOnDeactivation(&wl) || updated`
otherwise `ResetChecksOnDeactivation` may not be called if updated is already true.

Reply:
Ah, true. Thanks for catching 👍🏻